### DISCUSSION TITLE
Changing the Assignment Number Styling on the Schedule

### DISCUSSION THREAD

**Discussion by ajtruckle** [ajtruckle/meeting-schedule-assistant-support#32] [Q&A] [answered]
I received this question from a brother:

> Quiero preguntarte si es muy difícil hacer que el número de asignación sea en negrita y del color de la sección: Tesoros de la biblia, Seamos Mejores maestros y Nuestra vida cristiana

Translation according to DeepL:

> I want to ask you if it is too difficult to make the assignment number bold and in the colour of the section: Treasures of the Bible, Let's Be Better Teachers and Our Christian Life.

**Comment by ajtruckle** [comment, marked as answer]
Changing the style of the _assignment number_ can be done without too much difficulty. The instructions below assume that you are already using your own copy of one of the built-in templates.

### Bold ###
Locate the `@Index` template in your script and add the [`font-weight`](https://www.w3schools.com/cssref/pr_font_weight.php) property to the `span` object. Like this:

```xsl
<xsl:template match="@Index">
  <span style="font-size: 10pt; font-weight: bold;">
    <xsl:value-of select="."/>
    <xsl:text>. </xsl:text>
  </span>
</xsl:template>
```

### Text Colour ###
There are several ways to do this with XSL scripting. One way is to replicate the `@Index` template and make use of the `mode` property. Something like this:

```xsl
<xsl:template match="@Index" mode="TFGW">
  <span style="font-size: 10pt; font-weight: bold; color: #606a70;">
    <xsl:value-of select="."/>
    <xsl:text>. </xsl:text>
  </span>
</xsl:template>

<xsl:template match="@Index" mode="AYFM">
  <span style="font-size: 10pt; font-weight: bold; color: #c18626;">
    <xsl:value-of select="."/>
    <xsl:text>. </xsl:text>
  </span>
</xsl:template>

<xsl:template match="@Index" mode="LAC">
  <span style="font-size: 10pt; font-weight: bold; color: #961526;">
    <xsl:value-of select="."/>
    <xsl:text>. </xsl:text>
  </span>
</xsl:template>
```

Then, locate all instances of `<xsl:apply-templates select="@Index"/>` and adjust, given the context of that item:

- `<xsl:apply-templates select="@Index" mode="TFGW"/>`
- `<xsl:apply-templates select="@Index" mode="AYFM"/>`
- `<xsl:apply-templates select="@Index" mode="LAC"/>`

**Reply by MardoDan** [reply]
Después de realizar los cambios en la plantilla personalizada el resultado fue el esperado, aunque todavía tengo una duda:

![image](https://github.com/ajtruckle/meeting-schedule-assistant-support/assets/122133558/95c87bb1-cc9c-4d6c-a45a-423fbc235d9c)
Imagen 01

![image](https://github.com/ajtruckle/meeting-schedule-assistant-support/assets/122133558/3d9a4634-8982-43d5-82db-3ffc0a80949c)
Imagen 02

En la primera semana, cuando están las asignaciones de **Interés Sincero** (las asignaciones 4 y 5) no encuentro dónde modificar la plantilla para que el color sea el correcto tal como se muestra en la segunda imagen.

**Reply by ajtruckle** [reply]
@MardoDan 

> In the first week, when there are the Sincere Interest assignments (assignments 4 and 5) I cannot find where to modify the template so that the colour is correct as shown in the second image.

Those are referred to as `Teaching` items in the XML data. Look for this code block in the XSL script:

```xsl
<!-- Teaching assigning - 10 minute talk -->
<xsl:template match="Teaching">
	<tr>
		<xsl:apply-templates select="Time"/>
		<td class="cellTheme" colspan="2">
			<xsl:apply-templates select="@Index"/>
			<!--<span class="bulletAYFM">&#8226;</span>-->
			<span class="textTheme">
				<xsl:value-of select="Type"/>
			</span>
			<xsl:apply-templates select="Theme"/>
			<xsl:apply-templates select="Time" mode="Duration"/>
			<xsl:apply-templates select="SourceMaterial"/>
		</td>
		<xsl:apply-templates select="Names/Name"/>
	</tr>
</xsl:template>
```
Change `<xsl:apply-templates select="@Index"/>` to `<xsl:apply-templates select="@Index" mode="AYFM"/>`.

**Reply by MardoDan** [reply]
Funcionando Ok ahora. Cómo puedo hacer para  que el punto en Palabras de Introducción y Palabras de Conclusión tenga un espacio de 1mm al comienzo???

![imagen](https://github.com/ajtruckle/meeting-schedule-assistant-support/assets/122133558/d12add10-1acb-48e4-8d2f-87fe108c44c9)

![imagen](https://github.com/ajtruckle/meeting-schedule-assistant-support/assets/122133558/415b17c1-4085-417a-a57c-2fc521767c76)

**Reply by ajtruckle** [reply]
@MardoDan 
> Working Ok now. How can I make the point in Words of Introduction and Words of Conclusion have a space of 1mm at the beginning???

Please ask this as a new question. Thanks!

**Reply by MardoDan** [reply]
https://github.com/ajtruckle/meeting-schedule-assistant-support/issues/37
